File: android/src/main/kotlin/io/svnk/sato_printers/models/PrintResult.kt
package io.svnk.sato_printers.models

/**
 * Represents the result of a print operation.
 */
data class PrintResult(
    val success: Boolean,
    val message: String? = null,
    val responseData: ByteArray? = null
) {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "success" to success,
            "message" to message,
            "responseData" to responseData
        )
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as PrintResult

        if (success != other.success) return false
        if (message != other.message) return false
        if (responseData != null) {
            if (other.responseData == null) return false
            if (!responseData.contentEquals(other.responseData)) return false
        } else if (other.responseData != null) return false

        return true
    }

    override fun hashCode(): Int {
        var result = success.hashCode()
        result = 31 * result + (message?.hashCode() ?: 0)
        result = 31 * result + (responseData?.contentHashCode() ?: 0)
        return result
    }
}

File: lib/src/models/print_result.dart
import 'dart:typed_data';

/// Represents the result of a print operation.
class PrintResult {
  /// Whether the print operation was successful.
  final bool success;

  /// Optional message about the operation.
  final String? message;

  /// Response data from the printer, if any.
  final Uint8List? responseData;

  /// Creates a PrintResult.
  const PrintResult({required this.success, this.message, this.responseData});

  /// Creates a PrintResult from a map.
  factory PrintResult.fromMap(Map<dynamic, dynamic> map) {
    return PrintResult(
      success: map['success'] as bool? ?? false,
      message: map['message'] as String?,
      responseData: map['responseData'] != null
          ? Uint8List.fromList((map['responseData'] as List).cast<int>())
          : null,
    );
  }

  /// Creates a successful result.
  factory PrintResult.ok([String? message]) {
    return PrintResult(success: true, message: message);
  }

  /// Creates a failed result.
  factory PrintResult.error(String message) {
    return PrintResult(success: false, message: message);
  }

  @override
  String toString() => 'PrintResult(success: $success, message: $message)';
}
